tools/java/org/unicode/cldr/api/PathMatcher.java

```diff
@@ -0,0 +1,276 @@
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+package org.unicode.cldr.api;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkPositionIndex;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static org.unicode.cldr.api.AttributeKey.keyOf;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * An immutable matcher for {@link CldrPath} instances. A path matcher specification looks like
+ * {@code "//ldml/foo/*[@x="z"]/bar[@y=*]"}, where element names and attribute values can be
+ * wildcards. No capturing of wildcard values is done, and currently wildcards are limited to the
+ * entire field (element name, or attribute value).
+ *
+ * Examples of use:
+ *
+ * <h2>Matching an exact path</h2>
+ * <pre>{@code
```

Comment:
Do you need both pre and @code? I don't think I have seen the latter together with the former.
It definitely confuses GitHub comment highlighting.

Reply:
It's necessary since {@code} is what lets various meta-characters work and <pre> preserves multiline indents. Happy to change it, but it's the standard for JavaDoc in Google.